src/api/users/src/routes/users.js

```diff
@@ -29,112 +34,95 @@ router.get('/:id', validateId(), async (req, res, next) => {
 
 // get all users
 // rejected if the users collection is empty
-// otherwise n (specified via params) users are returned
+// otherwise perPage (specified via params) users are returned, starting at
+// the beginning of the users collection or the provided document path id (if present)
 router.get('/', validatePagingParams(), async (req, res, next) => {
   /*
-   *  celebrateSchema.js performs data validation via validatePagingParams() middleware
-   *  per_page is an integer validated to have a range between 1 and 20
-   *  page is an integer validated to have a min value of 1
+   *  schema.js performs data validation via validatePagingParams() middleware
+   *  per_page is an integer validated to have a range between 1 and 100
+   *  start_after is the id (hashed email) of the user to begin after
    */
-  const { per_page: perPage, page } = req.query;
-
-  // When requesting page 1 the user we start to build the query off of will be user 0
-  // when requesting page >= 2 the user we start at is determined by perPage * (page - 1)
-  const userToStartAt = page === 1 ? 0 : perPage * (page - 1);
-  const users = [];
+  const { per_page: perPage, start_after: startAfter } = req.query;
 
   try {
-    const query = await db
-      .collection('users')
-      .orderBy('id')
-      .startAt(userToStartAt)
-      .limit(perPage)
-      .get();
-
-    query.forEach((doc) => {
-      users.push(doc.data());
-    });
-
-    logger.debug(`Users were requested by ${req.ip} and served successfully.`);
+    let query = db.orderBy(documentId()).limit(perPage);
+
+    // If we were given a user ID to start after, use that document path to add .startAfter()
+    if (startAfter) {
+      query = query.startAfter(startAfter);
+    }
+
+    const snapshot = await query.get();
+    const users = snapshot.docs.map((doc) => doc.data());
+
+    // Add paging link header if necessary, so caller can request next page
+    addNextLinkHeader(res, users, perPage);
     res.status(200).json(users);
   } catch (err) {
     next(err);
   }
 });
 
-// post a user with supplied info, validated by the celebrateSchema
+// post a user with supplied info, validated by the schema
 // rejected if a user already exists with that id, otherwise user created
-router.post('/', validateUser(), async (req, res, next) => {
+router.post('/:id', validateId(), validateUser(), validateEmailHash(), async (req, res, next) => {
+  const { id } = req.params;
+  const { body } = req;
+
   try {
-    const userRef = db.collection('users').doc(`${req.body.id}`);
+    const userRef = db.doc(id);
     const doc = await userRef.get();
 
     if (doc.exists) {
-      logger.debug(`User (id: ${doc.id}) was posted by ${req.ip} but already exists.`);
-      res.status(400).json({
-        msg: `User with id ${doc.id} was requested to be added, but already exists in the db.`,
-      });
+      next(createError(400, `User with id ${id} already exists.`));
```

Comment:
```suggestion
      next(createError(400, `user with id ${id} already exists.`));
```

Might seem pedantic but `user` and `User` are both used throughout the PR. We should be more consistent throughout this PR.

Reply:
Fixed in https://github.com/Seneca-CDOT/telescope/pull/2060/commits/dc9a3d07b38f2b048b9caa13d9c5459a4c300f2a